classes/author/Collector.inc.php

```diff
@@ -179,12 +204,24 @@ public function getQuery()
                 });
         }
 
-        $q->orderBy('a.seq', 'ASC');
+        if (!empty($this->includeInBrowse)) {
+            $q->where('a.include_in_browse', $this->includeInBrowse);
+        }
+
+        if (!empty($this->count)) {
+            $q->limit($this->count);
+        }
+
+        if (!empty($this->offset)) {
+            $q->offset($this->offset);
+        }
+
+        $q->orderBy($this->orderBy, 'asc');
 
         // Add app-specific query statements
-        HookRegistry::call('Author::getMany::queryObject', [&$q, $this]);
+        HookRegistry::call('Author::Collector', [&$q, $this]);
 
-        $q->select($this->columns);
+        $q->distinct()->select($this->columns);
```

Comment:
I think @asmecher might have a recommendation here to avoid the use of `->distinct()`.

Reply:
Yes, a few patterns related to this:
- Remove `distinct()` by making any joins that multiply the results use a subquery, e.g.: https://github.com/pkp/pkp-lib/pull/7170/files#diff-1d3de820df407fc981c2a64acb5970be700de5a01a087ad65eab5719d7fdb133R394
- Remove the `columns` property; use a `main_table.*` selection pattern near the top of the query builder: https://github.com/pkp/pkp-lib/pull/7170/files#diff-1d3de820df407fc981c2a64acb5970be700de5a01a087ad65eab5719d7fdb133R364
- Any additional columns that need selecting should be added using `addSelect` near the join that brings them in: https://github.com/pkp/pkp-lib/pull/7170/files#diff-1d3de820df407fc981c2a64acb5970be700de5a01a087ad65eab5719d7fdb133R484

Reply:
Completely removed the `distinct` by removing the `leftjoins`